Flank/flank
# ci: Add build scan for ubuntu workflow

Fixes #1251 

Adds build scan to `ubuntu-workflow`. Additionally, commit ID will be added to the scan under `Custom Values`.
Currently only for `ubuntu-workflow`, if required, can be easily added to another GH actions

## Test Plan
> How do we know the code works?

1. Run `./gradlew build -I gradle/scans-init.gradle.kts`
1. Observe build scan URL is printed at the end of run
1. (to verify after merge) PR comment should be added when `ubuntu-workflow` is finished
